internal/pkg/service/stream/storage/node/coordinator/fileimport/operator.go

```diff
@@ -34,18 +37,20 @@ type operator struct {
 	clock        clock.Clock
 	logger       log.Logger
 	storage      *storageRepo.Repository
+	definition   *definitionRepo.Repository
 	statistics   *statsCache.L1
 	distribution *distribution.GroupNode
 	locks        *distlock.Provider
 	plugins      *plugin.Plugins
 
 	files *etcdop.MirrorMap[model.File, model.FileKey, *fileData]
+	sinks *etcdop.MirrorMap[definition.Sink, key.SinkKey, *sinkData]
 }
 
 type fileData struct {
-	FileKey model.FileKey
-	State   model.FileState
-	Retry   model.Retryable
+	model.FileKey
+	model.FileState
+	model.Retryable
```

Comment:
@jachym-tousek-keboola  Why did you use embedding here?

Reply:
But `FileState` for example has not method, ... and embeding is not intended to skip the field name only.

It is now a mixture of methods from 3 types, without any obvious intention/purpose.

@Matovidlo what do you think?

Reply:
I am not fan of embedding without reason. The reason behind embedding is to "inherit" methods of the embedded structures. However when we do not need them, therefore they should be present under specific key. I would rather consolidate it both to use the structures by name instead of inheriting it, just because we are lazy to write `data.FileKey` or `file.FileKey` .
```
type base struct {
    num int
}
func (b base) describe() string {
    return fmt.Sprintf("base with num=%v", b.num)
}

type container struct {
    base
    str string
}
```
Typicall usage is to use
```
 c := container{
        base: base{
            num: 1,
        },
        str: "some name",
    }
fmt.Println("describe:", c.describe())
```
If we do not use any method of embedded structures (or structure does not contain any method at all) it is pointless.